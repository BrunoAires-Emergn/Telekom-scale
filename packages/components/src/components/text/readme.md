# scale-text



<!-- Auto Generated Below -->


## Properties

| Property      | Attribute      | Description                    | Type                                     | Default     |
| ------------- | -------------- | ------------------------------ | ---------------------------------------- | ----------- |
| `customClass` | `custom-class` | (optional) Tag class           | `string`                                 | `''`        |
| `styles`      | --             | (optional) Injected jss styles | `StyleSheet<string \| number \| symbol>` | `undefined` |


----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
