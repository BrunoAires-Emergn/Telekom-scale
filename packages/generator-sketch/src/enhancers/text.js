const tinycolor = require("tinycolor2");

const enhancedStyle = (text, fontSize, fontFamily = "Arial-BoldMT", color) => ({
  _class: "style",
  endMarkerType: 0,
  miterLimit: 10,
  startMarkerType: 0,
  windingRule: 1,
  blur: {
    _class: "blur",
    isEnabled: false,
    center: "{0.5, 0.5}",
    motionAngle: 0,
    radius: 10,
    saturation: 1,
    type: 0
  },
  borderOptions: {
    _class: "borderOptions",
    isEnabled: true,
    dashPattern: [],
    lineCapStyle: 0,
    lineJoinStyle: 0
  },
  borders: [],
  colorControls: {
    _class: "colorControls",
    isEnabled: false,
    brightness: 0,
    contrast: 1,
    hue: 0,
    saturation: 1
  },
  contextSettings: {
    _class: "graphicsContextSettings",
    blendMode: 0,
    opacity: 1
  },
  fills: [],
  innerShadows: [],
  shadows: [],
  textStyle: {
    _class: "textStyle",
    encodedAttributes: {
      MSAttributedStringColorAttribute: {
        _class: "color",
        ...color
      },
      textStyleVerticalAlignmentKey: 0,
      MSAttributedStringFontAttribute: {
        _class: "fontDescriptor",
        attributes: {
          name: fontFamily,
          size: fontSize
        }
      },
      paragraphStyle: {
        _class: "paragraphStyle",
        alignment: 0
      },
      kerning: 0
    },
    verticalAlignment: 0
  }
});

const enhancedAttributedString = (
  text,
  fontSize,
  fontFamily = "Arial-BoldMT",
  color
) => ({
  _class: "attributedString",
  string: text,
  attributes: [
    {
      _class: "stringAttribute",
      location: 0,
      length: text.length,
      attributes: {
        MSAttributedStringFontAttribute: {
          _class: "fontDescriptor",
          attributes: {
            name: fontFamily,
            size: fontSize
          }
        },
        MSAttributedStringColorAttribute: {
          _class: "color",
          ...color
        },
        kerning: 0,
        textStyleVerticalAlignmentKey: 0,
        paragraphStyle: {
          _class: "paragraphStyle",
          alignment: 0
        }
      }
    }
  ]
});

const enhancedTypoStyle = (
  text,
  fontSize,
  fontFamily = undefined,
  color = undefined
) => ({
  style: enhancedStyle(text, fontSize, fontFamily, color),
  attributedString: enhancedAttributedString(text, fontSize, fontFamily, color)
});

function enhanceText(json) {
	return {
		...json,
		...enhancedTypoStyle(
			json.text,
			json.style.fontSize,
			json.style.fontFamily,
			json.style.color
		)
	}
}

module.exports = {
	enhanceText
}
