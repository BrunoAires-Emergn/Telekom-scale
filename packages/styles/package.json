{
    "name": "@telements/styles",
    "version": "0.0.1",
    "main": "dist/index.js",
    "scripts": {
        "build": "tsc -b",
        "postbuild": "node ./dist/export-css.js",
        "setup": "yarn"
    },
    "author": "@eeegor <egor.kirpichev@telekom.de>",
    "license": "MIT",
    "devDependencies": {
        "typescript": "^3.4.5"
    },
    "dependencies": {

    }
}
