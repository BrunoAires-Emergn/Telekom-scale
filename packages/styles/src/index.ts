import { button } from './button';
import { alert } from './alert';
import { card } from './card';

const theme = {
	colors: {
		white: '#fff',
		black: '#000',
		blue: '#00f',
		red: '#f00',
	}
}

export {
	theme,
	alert,
	button,
	card,
}
