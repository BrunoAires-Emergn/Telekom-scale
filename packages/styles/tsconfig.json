{
	"extends": "../tsconfig.base",
	"compilerOptions": {
		"rootDir": "./src",
		"outDir": "./dist",
		"types": [
			"node",
			"jest"
		]
	}
}
