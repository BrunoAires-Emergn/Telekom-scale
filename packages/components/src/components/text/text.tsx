import { Component, h, Prop, Host } from '@stencil/core';
import { CssClassMap } from '../../utils/utils';
import classNames from 'classnames';
import { styles } from './text.styles';
import { CssInJs } from '../../utils/css-in-js';
import { StyleSheet } from 'jss';
import Base from '../../utils/base-interface';

@Component({
  tag: 'scale-text',
  shadow: true,
})
export class Text implements Base {
  /** (optional) Tag class */
  @Prop() customClass?: string = '';

  /** (optional) Injected jss styles */
  @Prop() styles?: StyleSheet;
  /** decorator Jss stylesheet */
  @CssInJs('Text', styles) stylesheet: StyleSheet;

  componentWillLoad() {}
  componentWillUpdate() {}

  render() {
    return (
      <Host>
        <style>{this.stylesheet.toString()}</style>
        <div class={this.getCssClassMap()}>
          <slot />
        </div>
      </Host>
    );
  }

  getCssClassMap(): CssClassMap {
    const { classes } = this.stylesheet;
    return classNames(classes.text, this.customClass && this.customClass);
  }
}
