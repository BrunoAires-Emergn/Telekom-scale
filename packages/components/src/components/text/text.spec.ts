import { newSpecPage } from '@stencil/core/testing';
import { Text } from './text';
import { styles } from './text.styles';
import jss from 'jss';

describe('Text', () => {
  let element;
  let stylesheet;
  beforeEach(async () => {
    element = new Text();
    stylesheet = element.stylesheet = jss.createStyleSheet(styles as any);
  });

  it('should match snapshot', async () => {
    const page = await newSpecPage({
      components: [Text],
      html: `<t-text>Label</t-text>`,
    });
    expect(page.root).toMatchSnapshot();
  });

  // it('should handle a custom css class', () => {
  //   element.customClass = 'custom';
  //   expect(element.getCssClassMap()).toContain('custom');
  // });

  // it('should handle theme css class', () => {
  //   element.theme = 'default';
  //   expect(element.getCssClassMap()).toContain('text--theme-default');
  // });

  // it('should have a default css class', () => {
  //   expect(element.getCssClassMap()).toContain('text');
  // });
});
