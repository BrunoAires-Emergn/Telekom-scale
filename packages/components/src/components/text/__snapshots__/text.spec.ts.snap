// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text should match snapshot 1`] = `
<t-text>
  <mock:shadow-root>
    <style></style>
    <div class="text-0-0-2">
      <slot></slot>
    </div>
  </mock:shadow-root>
  Label
</t-text>
`;
